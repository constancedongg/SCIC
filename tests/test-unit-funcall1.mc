|'{km} = 0.001 '{m}|;

/* parameters can be auto-converted 
    when passed into function */
float '{km} func foo(float '{m} x) { 
    printf(x); /* 30 (m) */    
    float '{m} y = x;
    return y;   /* at return: convert to return unit km */
}

int func main() {
    float '{cm} tmp = 3000.0;
    printf(foo(tmp));  /* 0.03 (km)*/
    return 0;
}