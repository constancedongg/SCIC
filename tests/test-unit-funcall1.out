30
0.03